### PR TITLE
TestConfiguration - add OpenShift Online token property

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
@Crumby Please review

**Comment by tremes** [comment]
This needs to be fixed.

File: src/main/java/cz/xtf/TestConfiguration.java [hunk 1]
```diff
@@ -495,6 +499,8 @@ public static boolean openshiftOnline() {
 		return Boolean.parseBoolean(get().readValue(OPENSHIFT_ONLINE));
 	}
 
+	public static String openshiftOnlineToken() { return get().readValue(OPENSHIFT_ONLINE_TOKEN); }
```
Comment (Crumby) [inline src/main/java/cz/xtf/TestConfiguration.java]: There is already a property for token. See MASTER_TOKEN.

File: src/main/java/cz/xtf/openshift/OpenShiftUtils.java [hunk 2]
```diff
@@ -30,22 +31,39 @@ public static OpenShiftUtil admin(String namespace) {
...
+		        String namespace = TestConfiguration.masterNamespace();
+		        String username = TestConfiguration.masterUsername();
+		        String password = TestConfiguration.masterPassword();
 
-			masterUtil = getUtil(masterUrl, namespace, username, password);
+		        masterUtil = getUtil(masterUrl, namespace, username, password);
+		    } else {
+		        String namespace = TestConfiguration.masterNamespace();
+		        String masterUrl = TestConfiguration.masterUrl();
+		        masterUtil = new OpenShiftUtil(new OpenShiftConfigBuilder()
```
Comment (Crumby) [inline src/main/java/cz/xtf/openshift/OpenShiftUtils.java]: There is constructor in OpenShiftUtil that takes token instead username and password. Please use that one.

File: src/main/java/cz/xtf/openshift/OpenShiftUtils.java [hunk 3]
```diff
@@ -30,22 +31,39 @@ public static OpenShiftUtil admin(String namespace) {
...
-		String username = TestConfiguration.masterUsername();
-		String password = TestConfiguration.masterPassword();
-
-		return getUtil(masterUrl, namespace, username, password);
+		if(!TestConfiguration.openshiftOnline()) {
+		    String username = TestConfiguration.masterUsername();
+		    String password = TestConfiguration.masterPassword();
+		    return getUtil(masterUrl, namespace, username, password);
+		} else {
+		    return new OpenShiftUtil(new OpenShiftConfigBuilder()
```
Comment (Crumby) [inline src/main/java/cz/xtf/openshift/OpenShiftUtils.java]: new OpenShiftUtil(masterUrl, namespace, token);

File: src/main/java/cz/xtf/TestConfiguration.java [hunk 4]
```diff
@@ -178,7 +179,10 @@ public static TestConfiguration get() {
 	public static String masterUrl() {
 		String result = get().readValue(DOMAIN);
 		if (StringUtils.isNotBlank(result)) {
-			result = "https://" + (result.equals(CDK_DOMAIN) ? CDK_IP : ("api." + result)) + ":8443";
+			result = "https://" + (result.equals(CDK_DOMAIN) ? CDK_IP : ("api." + result));
+			if(!openshiftOnline()){
```
Comment (Crumby) [inline src/main/java/cz/xtf/TestConfiguration.java]: I don't believe that we should use xtf.config.domain property for OpenShift Online. Some other properties are delivered from this one (eg. ROUTE_DOMAIN, MAVEN_BASE_URL...) and could actually break a suite. MASTER_URL should be used with online testing

File: src/main/java/cz/xtf/TestConfiguration.java [hunk 5]
```diff
@@ -180,7 +180,7 @@ public static String masterUrl() {
 		if (StringUtils.isNotBlank(result)) {
 			result = "https://" + (result.equals(CDK_DOMAIN) ? CDK_IP : ("api." + result)) + ":8443";
 		} else {
-			result = get().readValue(MASTER_URL);
+			result = "https://" + "api." + get().readValue(MASTER_URL);
```
Comment (Crumby) [inline src/main/java/cz/xtf/TestConfiguration.java]: MASTER_URL property should already return the final url.

Comment (Crumby) [reply src/main/java/cz/xtf/TestConfiguration.java]: So if you want to test on online with token, you should set up test.properties with eg.:

xtf.config.master.url=https://api.openshift.domain.com
xtf.config.master.token=somerandomhash
xtf.config.master.namespace=myproject

**Pull request merged**